1026 allow subcategories for inst names

juzen2003 opened the pull request:
- Fixes #1026 
- Were any Django, import pipeline, or table_schema files modified? Y
  - If YES:
    - Database used: re-import
    - All Django tests pass: Y
    - FLAKE8 run on modified code: Y
- Were any JavaScript or CSS files modified? Y (CSS)
  - If YES:
    - JSHINT run on all affected files: NA
    - Tested on Chrome, Firefox, Safari, Edge, and iOS: NA
      (Remember to test all browser sizes)
    - Tested for race conditions (with delayed API calls if applicable): NA
- Was the documentation reviewed for necessary changes or additions? NA
  - About
  - Getting Started
  - FAQ
  - API Guide
  - Tooltips

Description of changes:
- Group ground based instruments into a subcategory ""Ground-based" in instrument widget.
- Add "Ground-based" to the mult_options of instrument_id field in obs_general.json.
- Update the logic to handle Django form when there is a mix of grouping and non-grouping checkboxes.

juzen2003 committed:
Group the ground-based instrument ids by updating form value passing to
Django form.
- In obs_general.json (line 77-82), add "Ground-based" to the last idx
  for row 17-22. This will make sure "Ground-based" being stored in the
  grouping column for all ground-based instruments in the
  mult_obs_general_instrument_id table.
- In config_data.py (line 27), add 'mult_obs_general_instrument_id' to
  MULT_TABLES_WITH_TARGET_GROUPING to indicate that the
  mult_obs_general_instrument_id table now has a grouping column.
- In mult_target_name_template.json (line 32), increase the size of
  grouping column so that long string like "Ground-based" can be stored.
- In views.py (line 431-441), add the logic to handle the form with a
  mix of grouping and non-grouping checkboxes. This type of form will
  be identified by (1) the existence of the grouping column and (2) None
  grouping value.

juzen2003 committed:
Add styling to left align grouping icon with non-grouping checkboxes in
the widget with a mix of grouping and non-grouping checkboxes.

juzen2003 committed:
Update the grouping size for these models: MultObsGeneralInstrumentId,
MultObsGeneralTargetName, and MultObsSurfaceGeometryNameTargetName.

juzen2003 committed:
Remove grouping_target_name table and update the target name mult field
function for ebrocc.py.
- Changes:
    - Remove the usage of mult_target_name_template.json
    - Remove MULT_TABLES_WITH_TARGET_GROUPING
    - Make sure grouping column exist in all mult tables
    - Remove grouping_target_name table
    - Remove ZZGroupingTargetName in search/models.py
    - Remove GroupingTargetName in metadata/models.py
    - Update the following mult field functions to return a dictionary:
        - In obs_general.py & obs_instrument_ebrocc.py:
            - field_obs_general_target_name
            - field_obs_general_target_class

juzen2003 committed:
Add grouping & group_disp_order for all mult tables. Mult fields with
preprogrammed options will be able to set the display order between
groups.
- Update all mult_options in .json files
- Remove mult_target_name_template.json

juzen2003 committed:
Fixed the missing ")" syntax error in search/models.py

juzen2003 committed:
Add the ability to specify disp_order and group_disp_order for non-preprogrammed
mult options. Make all the mult related field_* functions to return either
a string or a dictionary, and the return values will be used to update
the corresponding mult table.
- For mult tables with preprogrammed options:
    - add grouping column to store the group names
    - add group_disp_order column to store the group order
- For mult from field_* functions:
    - We return either a string or a dictionary with this structure:
        {
            'col_val': value for the mult option,
            'disp_name': display name on the UI,
            'disp_order': display order of this option within the same group, could be None
            'grouping': group name associated with this value, could be None
            'group_disp_order': display order of this group, could be None
        }
    - if disp_order or group_disp_order is not specify, they will be ordered alphabetically
- Update the following mult field functions:
    - In obs_general.py
        - field_obs_general_target_name
        - field_obs_general_target_class
    - In obs_instrument_ebrocc.py:
        - field_obs_general_target_name
        - field_obs_profile_quality_score
        - field_obs_profile_source
        - field_obs_profile_host

juzen2003 committed:
Remove MULT_TABLES_WITH_TARGET_GROUPING in config_data.py

juzen2003 committed:
Merge remote-tracking branch 'upstream/master' into 1026_allow_subcategories_for_inst_names

juzen2003 committed:
Add _create_mult function in obs_base.py to return a dictionary with all
the mult data. This function will be called for all the mult field
populate functions.
- Change TARGET_NAME_GROUP_MAPPING to PLANET_GROUP_MAPPING
- Update the followings to call _create_mult
    - In obs_general.py
        - field_obs_general_instrument_id
        - field_obs_general_inst_host_id
        - field_obs_general_mission_id
        - field_obs_general_target_class
        - field_obs_general_target_name
    - In obs_instrument_ebrocc.py
        - field_obs_general_planet_id
        - field_obs_general_target_name
        - field_obs_general_quantity
        - field_obs_general_observation_type
        - field_obs_profile_occ_type
        - field_obs_profile_occ_dir
        - field_obs_profile_body_occ_flag
        - field_obs_profile_quality_score
        - field_obs_profile_wl_band
        - field_obs_profile_source
        - field_obs_profile_host

juzen2003 committed:
Update the followings mult field_* functions to call _create_mult
    - field_obs_general_inst_host_id
    - field_obs_general_instrument_id
    - field_obs_general_mission_id
    - field_obs_general_observation_type
    - field_obs_general_planet_id

juzen2003 committed:
Update the followings mult field_* functions to call _create_mult
    - field_obs_general_quantity
    - field_obs_general_target_class
    - field_obs_general_target_name

juzen2003 committed:
Update the followings mult field_* functions to call _create_mult
    - field_obs_mission_voyager_mission_phase_name
    - field_obs_profile_host
    - field_obs_profile_body_occ_flag
    - field_obs_profile_occ_dir
    - field_obs_profile_occ_type
    - field_obs_profile_quality_score
    - field_obs_profile_source
    - field_obs_profile_wl_band

juzen2003 committed:
Update the followings mult field_* functions to call _create_mult
    - field_obs_surface_geometry_name_target_name
    - field_obs_type_image_image_type_id
    - field_obs_wavelength_polarization_type
    - field_obs_wavelength_spec_flag

juzen2003 committed:
- Update mult_options to have group_disp_order in the following files:
    - obs_general.json
    - obs_instrument_cocirs.json
    - obs_instrument_coiss.json
    - obs_instrument_couvis.json
    - obs_instrument_covims.json
    - obs_instrument_gossi.json
    - obs_instrument_nhlorri.json
    - obs_instrument_nhmvic.json
    - obs_instrument_vgiss.json
    - obs_mission_cassini.json
    - obs_mission_galileo.json
    - obs_mission_hubble.json
    - obs_profile.json
    - obs_wavelength.json
    - obs_type_image.json

juzen2003 committed:
- Update the followings field_* functions to call _create_mult:
    - field_obs_mission_cassini_cassini_target_code
    - field_obs_mission_cassini_cassini_target_name
    - field_obs_instrument_coiss_combined_filter
    - field_obs_mission_cassini_rev_no

juzen2003 committed:
- Update the field_.* functions with GROUP/TARGETS form type to call _create_mult
    - obs_general.json
    - obs_instrument_cocirs.json
    - obs_instrument_coiss.json
    - obs_instrument_couvis.json
    - obs_instrument_covims.json
    - obs_instrument_gossi.json
    - obs_instrument_nhlorri.json
    - obs_instrument_nhmvic.json
    - obs_instrument_vgiss.json
    - obs_mission_cassini.json
    - obs_mission_galileo.json
    - obs_mission_hubble.json
    - obs_mission_new_horizon.json
    - obs_mission_voyager.json
    - obs_profile.json
    - obs_surface_geometry_name.json
    - obs_wavelength.json
    - obs_type_image.json

juzen2003 committed:
Move the steps of handling the case when only column_val is set, and
rest of the mult data are None out of _create_mult.
- In do_import.py (line 1316-1331), handle the case when only column_val
  is set, and rest of the mult data are None. This stays here because
  mult_label gets updated based on column_val after column_val is validated.
  (ex: flag)

juzen2003 committed:
Remove parameter "key" in _create_mult function.

rfrenchseti commented on opus/import/config_targets.py:
```diff
@@ -923,3 +923,22 @@
+# For non-preprogrammed mult, if we want to specify display order or group
+# display order, we have to know all the possible options. For now, we put it
+# here, and will re-organize it later.
+# TODO: this grouping info is needed when we want to specify the order and
```
This comment can be removed

rfrenchseti commented on opus/import/main_opus_import.py:
```diff
@@ -488,8 +488,8 @@ def _new_warning_handler(message, category, filename, lineno, file, line):
             do_table_names.do_table_names()
-        if impglobals.ARGUMENTS.create_grouping_target_name:
-            do_grouping_target_name.do_grouping_target_name()
+        # if impglobals.ARGUMENTS.create_grouping_target_name:
```
Remove. Also delete do_grouping_target_name.py. Remove the --create-grouping-target-name option from main_opus_import and also remove all references to that argument from other argument help strings.

rfrenchseti commented on opus/import/obs_base.py:
```diff
@@ -343,6 +344,29 @@ def _get_target_info(self, target_name):
+                planet_id = 'OTHER'
+        return PLANET_GROUP_MAPPING[planet_id]
+
+    def _create_mult(
```
Change to put arguments on first line for consistency:
```
def _create_mult(self, col_val, col_class=None, disp_name=None,
                 disp_order=None, grouping=None, group_disp_order=None):
```

rfrenchseti commented on opus/import/obs_base.py:
```diff
@@ -343,6 +344,29 @@ def _get_target_info(self, target_name):
+        return PLANET_GROUP_MAPPING[planet_id]
+
+    def _create_mult(
+        self, col_val, col_class=None, disp_name=None,
```
Is `col_class` used for anything? If not, remove everywhere.

rfrenchseti commented on opus/application/apps/search/models.py:
```diff
@@ -162,24 +162,14 @@ class Meta:
     label = models.CharField(max_length=100)
     disp_order = models.CharField(max_length=100)
     display = models.CharField(max_length=1)
+    grouping = models.CharField(max_length=20, blank=True, null=True)
```
Fix grouping and grouping_disp_order to be the same

rfrenchseti commented on opus/import/obs_general.py:
```diff
@@ -125,8 +125,14 @@ def field_obs_general_target_name(self):
-        return target_name, target_info[2]
+            return self._create_mult(None)
+
+        data_dict = self._create_mult(
```
Collapse onto fewer lines, and you can just return directly instead of creating data_dict first.

rfrenchseti commented on opus/import/obs_instrument_cassini_occ.py:
```diff
@@ -26,16 +26,26 @@ def convert_filespec_from_lbl(self, filespec):
-        return target_name, target_info[2]
+        group_info = self._get_planet_group_info(target_name)
+
+        data_dict = self._create_mult(
```
Make more compact

rfrenchseti commented on opus/import/obs_instrument_corss_occ.py:
```diff
@@ -54,24 +54,37 @@ def field_obs_wavelength_wavelength2(self):
 
     def field_obs_profile_quality_score(self):
-        return ("UNASSIGNED", "Unassigned")
+        data_dict = self._create_mult(
```
Make compact and put with return statement directly

rfrenchseti commented on opus/import/obs_instrument_corss_occ.py:
```diff
@@ -54,24 +54,37 @@ def field_obs_wavelength_wavelength2(self):
 
     def field_obs_profile_source(self):
-        return ('CASSINI', 'Cassini', '!Cassini') # Force Cassini to be before all stars
+        data_dict = self._create_mult(
```
Make compact and put with return statement directly

rfrenchseti commented on opus/import/obs_instrument_corss_occ.py:
```diff
@@ -54,24 +54,37 @@ def field_obs_wavelength_wavelength2(self):
         dsn = self._supp_index_col('DSN_STATION_NUMBER')
         ret = f'DSN {dsn} ({DSN_NAMES[dsn]})'
-        return (ret, ret)
+        data_dict = self._create_mult(
```
Make compact and put with return statement directly

rfrenchseti commented on opus/import/obs_instrument_couvis_covims_occ.py:
```diff
@@ -49,19 +49,23 @@ def field_obs_pds_note(self):
-            return None
-        return target_name, target_name_info[2]
+            return self._create_mult(None)
+        data_dict = self._create_mult(
```
Make compact and put with return statement directly

rfrenchseti commented on opus/import/obs_instrument_covims.py:
```diff
@@ -318,7 +320,11 @@ def field_obs_instrument_covims_vis_exposure(self):
 
     def field_obs_instrument_covims_channel(self):
-        return (self.phase_name, self.phase_name)
+        data_dict = self._create_mult(
```
Make compact and put with return statement directly

rfrenchseti commented on opus/import/obs_instrument_ebrocc.py:
```diff
@@ -7,7 +7,7 @@
 
 from obs_common import ObsCommon
-
+from config_targets import *
```
We don't want to use "import *" anymore - list the specific things you need

rfrenchseti commented on opus/import/obs_instrument_ebrocc.py:
```diff
@@ -85,14 +85,23 @@ def field_obs_general_target_name(self):
+            return self._create_mult(None)
+        group_info = self._get_planet_group_info(target_name)
+
+        data_dict = self._create_mult(
```
Make more compact.

rfrenchseti commented on opus/import/obs_instrument_ebrocc.py:
```diff
@@ -111,38 +120,51 @@ def field_obs_wavelength_wavelength2(self):
 
     def field_obs_profile_quality_score(self):
-        return ("UNASSIGNED", "Unassigned")
+        data_dict = self._create_mult(
```
Make more compact. Also is this necessary. Doesn't "UNASSIGNED" automatically display as "Unassigned"? Original code might be unnecessary.

rfrenchseti commented on opus/import/obs_instrument_ebrocc.py:
```diff
@@ -111,38 +120,51 @@ def field_obs_wavelength_wavelength2(self):
-            return None
-        return target_name, target_info[2]
+            return self._create_mult(None)
+        data_dict = self._create_mult(
```
Make compact and combine with return

rfrenchseti commented on opus/import/obs_instrument_ebrocc.py:
```diff
@@ -111,38 +120,51 @@ def field_obs_wavelength_wavelength2(self):
     def field_obs_profile_host(self):
         insthost = self._supp_index_col('INSTRUMENT_HOST_NAME')
-        return (insthost, insthost)
+        data_dict = self._create_mult(
```
Make compact and combine with return

rfrenchseti commented on opus/import/obs_instrument_ebrocc.py:
```diff
@@ -111,38 +120,51 @@ def field_obs_wavelength_wavelength2(self):
         insthost = self._supp_index_col('INSTRUMENT_HOST_NAME')
-        return (insthost, insthost)
+        data_dict = self._create_mult(
+            col_val=insthost,
```
This would be a good place to have something like self._create_mult_keep_case(insthost)

rfrenchseti commented on opus/import/obs_instrument_hststis.py:
```diff
@@ -16,7 +16,7 @@ def __init__(self, *args, **kwargs):
 
     def _stis_spec_flag(self):
-        obs_type = self.field_obs_general_observation_type()
+        obs_type = self.field_obs_general_observation_type()['col_val']
```
This looks kind of ugly now. Perhaps we need a function `_observation_type` (put right above `field_obs_general_observation_type`) that does the work of the old `field_obs_general_observation_type` and just returns the raw value. That could be used in this function. Then the full `field_obs_general_observation_type` could call `_observation_type` and turn it into a mult with no additional work needed.

juzen2003 committed:
Remove the col_class parameter in the _create_mult function since it's
not used.

rfrenchseti commented on opus/import/obs_instrument_vg28xx_vgrss.py:
```diff
@@ -64,7 +70,8 @@ def field_obs_profile_host(self):
 
     def _is_voyager_at_uranus(self):
-        return self.field_obs_general_target_name()[0] == 'U RINGS'
+        return (self.field_obs_general_target_name()['col_val']
```
This can go back to being one line

rfrenchseti commented on opus/import/obs_mission_cassini.py:
```diff
@@ -415,24 +415,32 @@ def field_obs_mission_cassini_ert2(self):
         target_code = obs_parts[1][-2:]
         if target_code in _CASSINI_TARGET_CODE_MAPPING:
-            return target_code, _CASSINI_TARGET_CODE_MAPPING[target_code]
+            data_dict = self._create_mult(
```
One line, combine with return

rfrenchseti commented on opus/import/obs_surface_geometry_name.py:
```diff
@@ -34,5 +34,9 @@ def field_obs_surface_geometry_name_target_name(self):
-            return None
-        return target_name, target_info[2]
+            return self._create_mult(None)
+        data_dict = self._create_mult(
```
Make one line and combine

rfrenchseti commented on opus/import/table_schemas/mult_template.json:
```diff
@@ -27,6 +27,16 @@
     },
+    {
+        "field_name": "grouping",
+        "field_type": "char20",
```
Make 100

rfrenchseti commented on opus/import/table_schemas/mult_template.json:
```diff
@@ -27,6 +27,16 @@
+    {
+        "field_name": "group_disp_order",
+        "field_type": "char100",
+        "field_notnull": false
```
Fix null

rfrenchseti commented on opus/import/obs_instrument_vg28xx_vgrss.py:
```diff
@@ -28,34 +28,40 @@ def instrument_id(self):
 
         ret = f'DSN {dsn} ({DSN_NAMES[dsn]})'
-        return (ret, ret)
+        data_dict = self._create_mult(
```
Combine

juzen2003 committed:
Change length of grouping field to be char100 to allow longer grouping
name and also set default of both grouping and group_disp_order fields
to be null.
- In mult_template.json (line 32, 38), increase grouping field size to
  100 and set default value to null for group_disp_order.
- In search/models.py, change all grouping size to 100 and allow null to
  be the default value for group_disp_order.

juzen2003 committed:
Remove do_grouping_target_name.py and --create-grouping-target-name option
from import.

juzen2003 committed:
Replace _create_mult(col_val=x, disp_name=x) with _create_mult_keep_case(x)

juzen2003 committed:
Based on github comments, create helper functions _target_name,
_combined_filter, _observation_type, _mission_phase_name, _planet_id,
_rev_no, _prime_inst_id that will return raw values, and the related
field_* functions (for mult) will call them and pass the results to
_create_mult.

rfrenchseti commented on opus/import/obs_general.py:
```diff
@@ -31,22 +31,22 @@ def field_obs_general_volume_id(self):
-        if target_name is None:
-            return None
-        target_name, target_info = self._get_target_info(target_name[0])
+        if target_name['col_val'] is None:
```
Shoudn't this use the new `_target_name`?

juzen2003 replied:
Work in progress, need to override ```_take_name``` for instrument that doesn't derive target name from index file (```self._some_index_or_label_col('TARGET_NAME')```)

juzen2003 replied:
Fixed.

rfrenchseti commented on opus/import/obs_instrument_couvis.py:
```diff
@@ -151,23 +152,24 @@ def field_obs_general_ring_obs_id(self):
 
     def field_obs_general_planet_id(self):
-        return self._cassini_planet_id()
+        planet_id = self._cassini_planet_id()
```
Can make one line

rfrenchseti commented on opus/import/obs_instrument_coiss.py:
```diff
@@ -240,19 +240,21 @@ def field_obs_general_ring_obs_id(self):
 
     def field_obs_general_planet_id(self):
-        return self._cassini_planet_id()
+        planet_id = self._cassini_planet_id()
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_covims.py:
```diff
@@ -122,19 +123,20 @@ def field_obs_general_ring_obs_id(self):
 
     def field_obs_general_planet_id(self):
-        return self._cassini_planet_id()
+        planet_id = self._cassini_planet_id()
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_hstacs.py:
```diff
@@ -34,11 +34,15 @@ def instrument_id(self):
         return 'IMG'
 
+    def field_obs_general_observation_type(self):
+        obs_type = self._observation_type()
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_hststis.py:
```diff
@@ -42,6 +42,10 @@ def field_obs_general_observation_type(self):
         return 'IMG' # Image
 
+    def field_obs_general_observation_type(self):
+        obs_type = self._observation_type()
```
Can be one line

juzen2003 committed:
Remove unused FieldError in ui/view.py & mult_label_set in do_import.py.

rfrenchseti commented on opus/import/obs_instrument_hstwfc3.py:
```diff
@@ -36,11 +36,15 @@ def instrument_id(self):
         return 'IMG'
 
+    def field_obs_general_observation_type(self):
+        obs_type = self._observation_type()
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_hstnicmos.py:
```diff
@@ -33,12 +33,17 @@ def instrument_id(self):
 
 
+    def field_obs_general_observation_type(self):
+        obs_type = self._observation_type()
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_hstwfpc2.py:
```diff
@@ -28,9 +28,13 @@ def instrument_id(self):
         return 'IMG'
 
+    def field_obs_general_observation_type(self):
+        obs_type = self._observation_type()
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_vg28xx.py:
```diff
@@ -194,6 +194,11 @@ def field_obs_ring_geometry_ring_intercept_time2(self):
+        return mission_phase
+
+    def field_obs_mission_voyager_mission_phase_name(self):
+        mission_phase = self._mission_phase_name()
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_vg28xx.py:
```diff
@@ -194,6 +194,11 @@ def field_obs_ring_geometry_ring_intercept_time2(self):
+    def _mission_phase_name(self):
         target_name = self._index_col('TARGET_NAME').upper()
-        return _VG_TARGET_TO_MISSION_PHASE_MAPPING[target_name]
+        mission_phase = _VG_TARGET_TO_MISSION_PHASE_MAPPING[target_name]
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_vg28xx_vgpps_vguvs.py:
```diff
@@ -45,22 +45,25 @@ def field_obs_general_declination2(self):
 
     def field_obs_profile_occ_dir(self):
-        return self._index_col('RING_OCCULTATION_DIRECTION')[0]
+        occ_dir = self._index_col('RING_OCCULTATION_DIRECTION')[0]
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_vg28xx_vgpps_vguvs.py:
```diff
@@ -45,22 +45,25 @@ def field_obs_general_declination2(self):
 
     def field_obs_profile_body_occ_flag(self):
-        return self._supp_index_col('PLANETARY_OCCULTATION_FLAG')
+        occ_flag = self._supp_index_col('PLANETARY_OCCULTATION_FLAG')
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_vg28xx_vgpps_vguvs.py:
```diff
@@ -45,22 +45,25 @@ def field_obs_general_declination2(self):
     def field_obs_profile_host(self):
-        return self._supp_index_col('RECEIVER_HOST_NAME')
+        host = self._supp_index_col('RECEIVER_HOST_NAME')
+        return self._create_mult(host)
```
Can be one line

rfrenchseti commented on opus/import/obs_instrument_vgiss.py:
```diff
@@ -170,9 +170,13 @@ def field_obs_wavelength_wave_no_res2(self):
         return self._index_col('MISSION_PHASE_NAME')
 
+    def field_obs_mission_voyager_mission_phase_name(self):
+        mission_phase = self._mission_phase_name()
```
Can be one line

rfrenchseti commented on opus/import/obs_mission_cassini.py:
```diff
@@ -330,39 +330,43 @@ def field_obs_mission_cassini_instrument_id(self):
+
+    def field_obs_mission_cassini_rev_no(self):
+        rev_no = self._rev_no()
+        return self._create_mult_keep_case(rev_no)
```
Can be one line

rfrenchseti commented on opus/import/obs_mission_cassini.py:
```diff
@@ -400,6 +404,10 @@ def field_obs_mission_cassini_prime_inst_id(self):
 
+    def field_obs_mission_cassini_prime_inst_id(self):
+        prime_inst_id = self._prime_inst_id()
+        return self._create_mult(prime_inst_id)
```
Can be one line

rfrenchseti commented on opus/import/obs_mission_hubble.py:
```diff
@@ -60,32 +60,36 @@ def field_obs_general_ring_obs_id(self):
 
+    def field_obs_general_planet_id(self):
+        planet_id = self._planet_id()
+        return self._create_mult(planet_id)
```
Can be one line

rfrenchseti reviewed (changes requested):
Just a few minor stylistic changes. I'm running a full-database import to make sure there aren't any surprises.

juzen2003 committed:
Update code style based on github comments.

juzen2003 committed:
Override _target_name in cassini instruments and ebrocc.

juzen2003 committed:
Fixed the missing grouping issue on intended target name and surface
geometry target selector by passing in grouping and group_disp_order
info to _create_mult in field_obs_general_target_name and
field_obs_surface_geometry_name_target_name.

juzen2003 committed:
For mission hubble, group filter name and aperture by its instrument name.
- In obs_mission_hubble.py (line 215,224), passed in instrument name as
  the grouping parameter for both filter name and aperture fields.

juzen2003 committed:
- Group HST filter by instrument names.
- Group HST aperture type by instrument names.
- Group part of OCC receiver host by DSNs and Ground-based Telescopes.
- Group part of OCC signal source by Stars

juzen2003 committed:
Add 'Venus' and 'Mercury' to PLANET_GROUP_MAPPING.

juzen2003 committed:
Update _create_mult_keep_case to also take in disp_order, grouping=None,
and group_disp_order as parameters. Make mult display value to be the
title case if it's not specified.
